src/sqs/sqs.ts

```diff
@@ -267,8 +271,12 @@ export class SqsQueue<MessagePayload> {
 		try {
 			const messageProcessingStartTime = Date.now();
 
-			if (await preemptiveRateLimitCheck(context, this)) {
-				context.log.info("Preemptive rate limit threshold exceeded");
+			const rateLimitCheckResult = await preemptiveRateLimitCheck(context, this);
+			if (rateLimitCheckResult.isExceedThreshold) {
+				//threshold exceeded, retry via sending a new message.
```

Comment:
nit: comment duplicating code

Reply:
I'd rather explain why we are doing this, not what we are doing

Reply:
fixed.